heron/schedulers/src/java/com/twitter/heron/scheduler/kubernetes/KubernetesController.java

```diff
@@ -109,44 +233,13 @@ public boolean submitTopology(String[] appConfs) {
     boolean allSuccessful = true;
 
     for (int i = 0; i < appConfs.length; i++) {
-      LOG.log(Level.INFO, "Topology configuration is: " + appConfs[i]);
 
-      // Get a connection
-      HttpURLConnection conn = NetworkUtils.getHttpConnection(deploymentURI);
-      if (conn == null) {
-        LOG.log(Level.SEVERE, "Fauled to find k8s deployment API");
-        return false;
+      allSuccessful = deployContainer(appConfs[i]);
```

Comment:
You can remove the need for `allSuccessful` by doing this here and returning true at the end:
```
if (!deployContainer(appConfs[i])) {
  return false;
}
```

Or better yet, change this method to return void and throw an exception if one fails. It can have detailed info included about what went wrong.